src/config/config.c

```diff
@@ -410,21 +410,20 @@ cfgOptionIdxTotal(const ConfigOption optionId)
 
 /**********************************************************************************************************************************/
 FN_EXTERN const String *
-cfgOptionDefault(const ConfigOption optionId)
+cfgOptionIdxDefaultValue(const ConfigOption optionId, const unsigned int optionIdx)
 {
     FUNCTION_TEST_BEGIN();
         FUNCTION_TEST_PARAM(ENUM, optionId);
+        FUNCTION_TEST_PARAM(UINT, optionIdx);
     FUNCTION_TEST_END();
 
     ASSERT(cfgInited());
     ASSERT(optionId < CFG_OPTION_TOTAL);
+    ASSERT_DECLARE(const bool group = configLocal->option[optionId].group);
+    ASSERT_DECLARE(const unsigned int indexTotal = configLocal->optionGroup[configLocal->option[optionId].groupId].indexTotal);
```

Comment:
This is a little hard to follow; not sure if there's anything to do to make it a little more clear, but maybe some sort of macro?

Reply:
(Though at that point, you're just hiding that complexity somewhere else for me to complain about, so maybe this is already as isolated as we can hope for...)

Reply:
Yeah, that was my thinking. I originally tried to create wrapper macros but there are subtle differences per function so they were impossible to maintain or understand.